### PR TITLE
Add support for screen_hint=signup param

### PR DESCRIPTION
**Pull request by bbean86**
### Changes
This adds support for the Hosted Login `screen_hint=signup` parameter. I discovered this issue while implementing Auth0 in a new Rails 6 app using Hosted Login. I noticed others had the same experience, so I thought I'd share my solution.

### References

https://github.com/auth0/omniauth-auth0/issues/87

### Testing

This can be tested by adding the `screen_hint=signup` query parameter to the `/auth/auth0` request. You should be able to observe the `screen_hint` parameter being sent to the Auth0 `/authorize` endpoint.

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [x] I have read the [Auth0 contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [x] All existing and new tests complete without errors
* [x] All code quality tools/guidelines in the [CONTRIBUTING documentation](https://github.com/auth0/omniauth-auth0/blob/master/CONTRIBUTING.md) have been run/followed

### CODE REVIEW

File: lib/omniauth/strategies/auth0.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ def client
       def authorize_params
         params = super
         parsed_query = Rack::Utils.parse_query(request.query_string)
-        %w[connection connection_scope prompt].each do |key|
+        %w[connection connection_scope prompt screen_hint].each do |key|
```
Comment (lbalmaceda) [inline lib/omniauth/strategies/auth0.rb:89]: what is this line doing, allowing these keys to be taken from the URL query and setting them on the /authorize call?

File: spec/omniauth/strategies/auth0_spec.rb [hunk 2]
```diff
@@ -123,6 +123,20 @@
       expect(redirect_url).not_to have_query('connection')
     end
 
+    it 'redirects to hosted login page with screen_hint=signup' do
+      get 'auth/auth0?screen_hint=signup'
```
Comment (lbalmaceda) [inline spec/omniauth/strategies/auth0_spec.rb:127]: There's a existing test for checking that on a "normal authorize call" (without extra args) these keys are not part of the query. Let's add this one there, and also the ones that are missing from previous changes.

The test is on these lines: https://github.com/auth0/omniauth-auth0/blob/894d4e29046fb2944119c2fa462ef6aa4436fd62/spec/omniauth/strategies/auth0_spec.rb#L84-L86

Let's add a line for testing that the following keys are NOT in the query:
- connection_scope
- screen_hint

**Comment by lbalmaceda** [comment]
👋  @bbean86 friendly check-in, are you still interested in these changes?

**Comment by bbean86** [comment]
@lbalmaceda apologies, I will circle back to these updates this weekend. This update is still relevant to my needs.

**Review by lbalmaceda — APPROVED** [review]

**Comment by lbalmaceda** [comment]
@bbean86 thanks for the update! My team is tracking the pending release internally under `SDK-1966`.

**Pull request merged**